Tella/Scenes/Uwazi/ViewModel/DraftUwaziEntity.swift

```diff
@@ -14,27 +14,77 @@ class DraftUwaziEntity: ObservableObject {
     
     @Published var template: CollectedTemplate
     @Published var text: String = ""
-    
+    @Published var entryPrompts: [UwaziEntryPrompt] = []
+
     // Fields validation
     @Published var shouldShowError : Bool = false
     @Published var isValidText : Bool = false
-    
     @Published var propertyValues : [String:Any] = [ : ]
-    
-    func initializePropertyTextValues() {
+
+    let uwaziTitleString = "title"
+
+    init(mainAppModel: MainAppModel, template: CollectedTemplate) {
+        self.mainAppModel = mainAppModel
+        self.template = template
+        handleEntryPrompts()
+        dump(template.entityRow?.properties)
+        dump(propertyValues)
+    }
+
+
+    func handleEntryPrompts() {
         // Iterate through the template's properties and set initial values in propertyTextValues
-        for property in template.entityRow?.properties ?? [] {
-            if let label = property.label {
-                propertyValues[label] = ""
-            }
-        }
-        
-        for commonProperty in template.entityRow?.commonProperties ?? [] {
-            if let label = commonProperty.label {
-                propertyValues[label] = ""
-            }
-        }
+        handlePdfsPrompt()
+        handleSupportPrompt()
+        handleTitlePrompt()
+        handleEntryPromptForProperties()
     }
+
+
+    fileprivate func handlePdfsPrompt() {
```

Comment:
I think this shouldn't be inside the view model because it is mostly for parsing the different types of entity. Maybe moving it into a UwaziParser class

Reply:
Its a initial draft but I was going to move it to a different class anyways
Thanks for the suggestion